src/site-server.ts

```diff
@@ -335,45 +339,123 @@ export class SiteServer {
 		} = {}
 	): Promise< WpCliResult > {
 		const projectPath = this.details.path;
-		const phpVersion = targetPhpVersion ?? this.details.phpVersion;
 
-		if ( ! this.wpCliExecutor ) {
-			this.wpCliExecutor = new WpCliProcess( projectPath );
-			await this.wpCliExecutor.init();
+		// If args is a string, parse it with shell-quote. If it's an array, use directly.
+		let wpCliArgs: string[];
+		if ( typeof args === 'string' ) {
+			const parsedArgs = parse( args );
+
+			// The parsing of arguments can include shell operators like `>` or `||` that the app don't support.
+			const isValidCommand = parsedArgs.every(
+				( arg: unknown ) => typeof arg === 'string' || arg instanceof String
+			);
+			if ( ! isValidCommand ) {
+				return Promise.resolve( {
+					stdout: '',
+					stderr: `Cannot execute wp-cli command with arguments: ${ args }`,
+					exitCode: 1,
+				} );
+			}
+			wpCliArgs = parsedArgs as string[];
+		} else {
+			wpCliArgs = args;
 		}
 
-		const wpCliArgs = parse( args );
+		const cliArgs: string[] = [ 'wp', '--path', projectPath ];
```

Comment:
```suggestion
		const cliArgs: string[] = [ 'wp', '--path', this.details.path ];
```
Nit, but I don't think `projectPath` is used anywhere else, so we can probably remove it.